src/inc/macros.h

```diff
@@ -7,3 +7,8 @@
 #ifndef ExpectedPtr
 #define ExpectedPtr(ptr) { if (nullptr == ptr) return E_POINTER; }
 #endif
+
+#ifndef HRESULT_FROM_ERRNO
+#define HRESULT_FROM_ERRNO(value) \
+        MAKE_HRESULT(value == 0 ? SEVERITY_SUCCESS : SEVERITY_ERROR, FACILITY_NULL, HRESULT_CODE(value))
```

Comment:
Not sure if this is a common pattern, but based on the usage of this macro the ternary seems redundant - the caller appears to only use this when `errno != 0`, thus, the condition should never evaluate to true. That being said, I don't know if there's a future use-case where we would want to use `SEVERITY_SUCCESS`; if there isn't, should we enforce that `value` cannot be 0 here so that the caller can't inadvertently pass an `errno` that equals 0?

Reply:
This should be fine the way it is. If a future caller forgets to check that errno != 0, then it will effectively return `(HRESULT)0`, which should be the same as `S_OK`.